smithy-typescript-codegen/src/test/java/software/amazon/smithy/typescript/codegen/StructureGeneratorTest.java

```diff
@@ -43,7 +43,92 @@ public void properlyGeneratesRequiredMessageMemberOfException() {
                                   + "}");
     }
 
-    public void testErrorStructureCodegen(String file, String expectedType) {
+    @Test
+    public void filtersSensitiveSimpleShape() {
+        testStructureCodegen("test-sensitive-simple-shape.smithy",
+                                "  export const filterSensitiveLog = (obj: GetFooInput): any => ({\n"
+                                + "    ...obj,\n"
+                                + "    ...(obj.password && { password:\n"
+                                + "      SENSITIVE_STRING\n"
+                                + "    }),\n"
+                                + "  })\n");
+    }
+
+    @Test
+    public void callsFilterForStructureWithSensitiveData() {
```

Comment:
None of the `*WithSensitiveData` tests assert that the generated type (`User`) has the `@sensitive` component filtered. You can capture the response of `testStructureCodegen()` and make additional assertions against it.

Reply:
Added tests using the same mock models:
* `callsFilterInStructureWithSensitiveData`
* `callsFilterInListWithSensitiveData`
* `callsFilterInMapWithSensitiveData`

Commit [commit](https://github.com/awslabs/smithy-typescript/pull/170/commits/c8fced3c2c4b92133f7853d5907218de0f54b195)